### PR TITLE
[python-experimental] Fix model tests

### PR DESCRIPTION
**Pull request by timgclark**
* Fixes model tests by templating the classname instead of
the model's description.
* For the v3 sample only, creates a tests_manual folder and moves manually written tests into them. All tests in the test folder are now autogenerated and may be deleted and generated again with no coverage loss

### PR checklist
 
- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [ ] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [ ] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by timgclark** [comment]
@spacether

**Comment by wing328** [comment]
Restarted the failed CI jobs. Let's see how that goes.

File: bin/configs/python-experimental.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 generatorName: python-experimental
 outputDir: samples/openapi3/client/petstore/python-experimental
 inputSpec: modules/openapi-generator/src/test/resources/3_0/python-experimental/petstore-with-fake-endpoints-models-for-testing-with-http-signature.yaml
-templateDir: modules/openapi-generator/src/main/resources/python
```
Comment (spacether) [inline bin/configs/python-experimental.yaml:4]: Python-experimental uses some python templates to generate python code and mostly uses the templates one level deeper.
Why have you changed this?

File: modules/openapi-generator/src/main/resources/python/python-experimental/model_test.mustache [hunk 2]
```diff
@@ -11,22 +11,22 @@ import {{packageName}}
...
     def tearDown(self):
         pass
 
-    def test{{unescapedDescription}}(self):
-        """Test {{unescapedDescription}}"""
+    def test{{classname}}(self):
+        """Test {{classname}}"""
         # FIXME: construct object with mandatory attributes with example values
-        # model = {{unescapedDescription}}()  # noqa: E501
+        # model = {{classname}}()  # noqa: E501
```
Comment (spacether) [inline modules/openapi-generator/src/main/resources/python/python-experimental/model_test.mustache:29]: Thank you for this fix! I missed this when I switched python-experimental to using classname.

File: samples/openapi3/client/extensions/x-auth-id-alias/python-experimental/setup.py [hunk 3]
```diff
@@ -22,10 +22,10 @@
...
 
 REQUIRES = [
-  "urllib3 >= 1.15",
-  "certifi",
-  "python-dateutil",
-  "nulltype",
+    "urllib3 >= 1.15",
+    "certifi",
+    "python-dateutil",
+    "nulltype",
```
Comment (spacether) [inline samples/openapi3/client/extensions/x-auth-id-alias/python-experimental/setup.py:28]: Any chance of moving your black updates into a separate PR?
It will make it easier to review.

File: samples/openapi3/client/petstore/python-experimental/test/test_drawing.py [hunk 4]
```diff
@@ -41,130 +34,12 @@ def setUp(self):
     def tearDown(self):
         pass
 
-    def test_create_instances(self):
```
Comment (spacether) [inline samples/openapi3/client/petstore/python-experimental/test/test_drawing.py:44]: So some of the files in the tests directory have real tests.
We see that this method test_create_instances has a manually written test in it.
This update is deleting those real tests.
Please preserve them?

File: samples/openapi3/client/extensions/x-auth-id-alias/python-experimental/x_auth_id_alias/api/usage_api.py [hunk 5]
```diff
@@ -77,78 +74,61 @@ def __any_key(
...
+                        str: (
+                            bool,
+                            date,
+                            datetime,
+                            dict,
+                            float,
+                            int,
+                            list,
+                            str,
+                            none_type,
```
Comment (spacether) [inline samples/openapi3/client/extensions/x-auth-id-alias/python-experimental/x_auth_id_alias/api/usage_api.py:101]: I see this change in the samples but not in the mustache file.
Did you run black on the sample results manually?
If so CI will fail because the file contents disagree with the CI file contents.
If we want these black changes, they must be applied to our mustache files to have them show up in the samples.

File: samples/openapi3/client/extensions/x-auth-id-alias/python-experimental/setup.py (hunk 3 above)
Comment (timgclark) [reply samples/openapi3/client/extensions/x-auth-id-alias/python-experimental/setup.py:28]: Yes - black updates were accidental, will pull those out.

File: samples/openapi3/client/petstore/python-experimental/test/test_drawing.py (hunk 4 above)
Comment (timgclark) [reply samples/openapi3/client/petstore/python-experimental/test/test_drawing.py:44]: Let me take a stab at cleaning this up as well.

I want to open a seperate core issue around this in general.  The fact that we both commit the model tests and don't regenerate model tests when present masked this issue for me.  (This is neither a huge priority, nor is there a single obvious fix for it)

File: samples/openapi3/client/extensions/x-auth-id-alias/python-experimental/x_auth_id_alias/api/usage_api.py (hunk 5 above)
Comment (timgclark) [reply samples/openapi3/client/extensions/x-auth-id-alias/python-experimental/x_auth_id_alias/api/usage_api.py:101]: Yes, see above on accidentally committing the black changes. Fixing this up.

File: samples/openapi3/client/petstore/python-experimental/test/test_drawing.py (hunk 4 above)
Comment (spacether) [reply samples/openapi3/client/petstore/python-experimental/test/test_drawing.py:44]: Can I suggest a fix? This has come up a couple of times.
How about:
- Moving these manually written tests into the tests folder
- Renaming the tests folder something like tests_manual

File: bin/configs/python-experimental.yaml (hunk 1 above)
Comment (timgclark) [reply bin/configs/python-experimental.yaml]: This was just misunderstanding on my part.  That it's overridden within the generator makes perfect sense.

File: samples/openapi3/client/extensions/x-auth-id-alias/python-experimental/setup.py (hunk 3 above)
Comment (timgclark) [reply samples/openapi3/client/extensions/x-auth-id-alias/python-experimental/setup.py]: Reverted.

File: samples/openapi3/client/petstore/python-experimental/test/test_drawing.py (hunk 4 above)
Comment (timgclark) [reply samples/openapi3/client/petstore/python-experimental/test/test_drawing.py]: I pulled out all but one of the sample test changes.  Seeing that we have both the generated test stubs and real tests, I'm not sure how much value there is in keeping all of the stubs up to date.

File: samples/openapi3/client/extensions/x-auth-id-alias/python-experimental/x_auth_id_alias/api/usage_api.py (hunk 5 above)
Comment (timgclark) [reply samples/openapi3/client/extensions/x-auth-id-alias/python-experimental/x_auth_id_alias/api/usage_api.py]: Reverted.

File: samples/openapi3/client/petstore/python-experimental/test/test_drawing.py (hunk 4 above)
Comment (timgclark) [reply samples/openapi3/client/petstore/python-experimental/test/test_drawing.py]: Missed that last round of suggestions - let me take a run at that.

Comment (timgclark) [reply samples/openapi3/client/petstore/python-experimental/test/test_drawing.py:44]: Moved `tests` to `tests_manual`.

Moved the following from `test` into `tests_manual`:
> samples/openapi3/client/petstore/python-experimental/test/test_drawing.py
> samples/openapi3/client/petstore/python-experimental/test/test_fake_api.py
> samples/openapi3/client/petstore/python-experimental/test/test_shape.py
> samples/openapi3/client/petstore/python-experimental/test/test_fruit_req.py
> samples/openapi3/client/petstore/python-experimental/test/test_gm_fruit.py
> samples/openapi3/client/petstore/python-experimental/test/test_fruit.py

Regenerated all tests in the `test` directory.

Squashed everything down to one commit - I hated to do it, but the reason some of the tests were generating incorrectly was drift from master since I started working on this more than a week back.

File: bin/configs/python-experimental.yaml (hunk 1 above)
Comment (spacether) [reply bin/configs/python-experimental.yaml:4]: So can you revert this change ;p?

Comment (timgclark) [reply bin/configs/python-experimental.yaml]: I saw this came back in along with the black changes - this was easy, but took some doing to get the diff happy on black.

File: samples/openapi3/client/petstore/python-experimental/test/test_drawing.py (hunk 4 above)
Comment (spacether) [reply samples/openapi3/client/petstore/python-experimental/test/test_drawing.py:44]: Thank you very much!

**Comment by timgclark** [comment]
> Can you also move over these files?

Thanks for spelling out the missing tests. I suppose a search on `assert` would've been more effective than poorly parsing the diff :+1:
